docs/src/components/AlgoliaSearch/searchResult.js

```diff
@@ -0,0 +1,267 @@
+                <div className="px-4">
+                  <Link to={`/${item.link}`} className="search-result-link">
+                    <div className="p-4 search-result-entry d-flex align-items-center">
+                      <Icon className="mx-4" appearance="subtle" size={16} name="subdirectory_arrow_right" />
+                      <Text dangerouslySetInnerHTML={{ __html: getHighlightedText(item.name, SearchQuery) }} />
+                    </div>
+                  </Link>
+                </div>
+              )}
+            </div>
+          );
+        })}
+    </div>
+  );
+};
+
+const ShowResults = ({ list }) => {
+  let PageName = '';
+  const isSamePage = (data) => {
+    if (pageName(data) !== PageName) {
+      PageName = pageName(data);
+      return true;
+    }
+    return false;
+  };
+  return (
+    <div>
+      {list.map(
+        (data) =>
+          data.slug && (
+            <div>
+              {isSamePage(data) && (
+                <Subheading className="pb-4 px-6 search-category" appearance="subtle">
+                  {pageName(data)}
+                </Subheading>
+              )}
+              <CustomResultEntry data={data} />
+            </div>
+          )
+      )}
+    </div>
+  );
+};
+
+const pageName = (data) => {
+  const slugPath = data.slug.split('/');
+  let category = '';
+  if (slugPath[0] === 'mobile') {
+    category = slugPath[1];
+  } else {
+    category = slugPath[0];
+  }
+  return category;
+};
+
+const ErrorTemplate = () => {
+  return (
+    <EmptyState
+      size="small"
+      description="Sorry! We could not find any match. Please try again"
+      imageSrc={noResultImage}
+      title={`No results found for '${SearchQuery}'`}
+      className="errorTemplate"
+    />
+  );
+};
+
+const PageHit = ({ hits }) => {
+  const [selectedTab, setSelectedTab] = React.useState(0);
+
+  if (hits.length == 0) {
+    return <ErrorTemplate />;
+  }
+  const components = [],
+    patterns = [],
+    foundations = [],
+    content = [],
+    resources = [],
+    introduction = [],
+    allContent = [],
+    pagePriority = [];
+
+  hits.forEach((item) => {
+    let PageName = '';
+    if (item.slug) {
+      PageName = pageName(item);
+    }
+    if (item.slug && !pagePriority.includes(PageName)) {
+      if (item.title.toLowerCase().includes(SearchQuery.toLowerCase())) {
+        pagePriority.push(PageName);
+      } else if (item.headings.findIndex((heading) => heading.toLowerCase().includes(SearchQuery.toLowerCase())) > 0) {
+        pagePriority.push(PageName);
+      } else if (item.description.toLowerCase().includes(SearchQuery.toLowerCase())) {
+        pagePriority.push(PageName);
+      } else if (item.keywords && item.keywords.toLowerCase().includes(SearchQuery.toLowerCase())) {
+        pagePriority.push(PageName);
+      } else if (item.parentTab && item.parentTab.toLowerCase().includes(SearchQuery.toLowerCase())) {
+        pagePriority.push(PageName);
+      }
```

Comment:
What is the use of these conditions when we are inserting everything into the same array?